docs/proposals/unified-backend-architecture/tech-spec.md

```diff
@@ -0,0 +1,221 @@
+to handle information about `C`'s wallet(s), because `wallet-db` is storing in the cluster.
+
+### Wallet Import/Export
+
+Wallet can be imported using client's secret key.
+
+**TODO**: To clarify this process. User can be able to obtain its secret key.
+
+### Wallet Restoring
+
+Wallet can be restored. Use case: user changed a smartphone, install mobile Daedalus client and restore its wallet
+using backup phrase (for example, 12 words) generated during this wallet creation.
+
+Wallet restoring produces a `POST`-request to `/api/v1/wallets/?`, and node checks if such a wallet exists. If so,
+response contains an information about the wallet.
+
+**TODO**: To clarify this process.
+
+### View Balance
+
+User can see its current balance (by default, the balance should always be shown in the client application).
+
+After client application launching and after transactions client is asking for current balance. It produces
+`GET`(?)-request to `/api/v1/?`, and response contains an information about current balance on particular wallet.
+
+Please note that user can have few wallets with different balances, so balance for current active wallet will be
+asked and shown.
+
+**TODO**: To clarify endpoints.
+
+### Receive a Payment
+
+User can receive a payment from other user. To do it he must provide a receive address.
+
+**TODO**: To clarify details of receive address generation. Will it be generated on the client side on on the node
+side? If latter - we need additional API endpoint, something like `/api/v1/address/receive/new`, to obtain new
+receive address.
+
+### Send a Payment
+
+There are 3 steps to make a new payment:
+
+1. create transaction,
+2. sign transaction,
+3. publish transaction.
+
+When user creates a new payment (by defining recipient's address, amount and fee), it produces a `POST`-request
+to `/api/v1/transactions/unsigned`, and response will contain a new transaction without a witness (signature).
+
+After that user signs new transaction using its private keys (hosted on the client side).
+
+Then user send a payment. It produces a `POST`-request to `/api/v1/transactions/signed`, node receives transaction
+with a signature, forms transaction witness and publish this transaction to the Cardano network as usually.
+
+## API Extension
+
+API v1 must be extended to support both the mobile clients and the hardware wallets.
+
+### Create Unsigned Transaction
+
+When user wants to send money to other user, he creates a new payment. It produces a `POST`-request
+to `/api/v1/transactions/unsigned` endpoint, and response will contain a new transaction without a
+witness (signature).
+
+### Send Signed Transaction
+
+After user signs new transaction using its private keys (hosted on the client side) he sends it with
+a signature. It produces a `POST`-request to `/api/v1/transactions/signed`, node receives transaction
+with a signature, forms transaction witness and publish this transaction as usually.
+
```

Comment:
@adinapoli-iohk Do we need some additional API endpoints for mobile clients?

Reply:
Not AFAIK, this should be the same flow for hardware or mobile clients.